source/server/listener_impl.cc

```diff
@@ -427,9 +525,11 @@ ListenerImpl::~ListenerImpl() {
     // parent's initManager to get ready().
     listener_init_target_.ready();
   }
+  // destruct immediately to prevent other member destruction triggered the ready().
```

Comment:
nit: Start comment with capital end with period, please check the rest of the PR.

Reply:
Also, why is this needed? This is in the destructor. Can you just reorder the members if that matters?

Reply:
Will do. Add a TODO here to avoid too many things in this PR